tools/rostest/src/rostest/runner.py

```diff
@@ -190,7 +190,9 @@ def setUp(self):
     # new test_parent for each run. we are a bit inefficient as it would be possible to
     # reuse the roslaunch base infrastructure for each test, but the roslaunch code
     # is not abstracted well enough yet
-    self.test_parent = ROSTestLaunchParent(self.config, [self.test_file])
+
+    # As long as passed here the port num can be specified.
+    self.test_parent = ROSTestLaunchParent(self.config, [self.test_file], self.config.master.get_port())
```

Comment:
Please pass keyword arguments with the keyword and not as a position argument (`port=...`).